src/FormInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Form;
+
+interface FormInterface
+{
+    public function addError(string $attribute, string $error): void;
+    public function addErrors(array $items): void;
+    public function attributes(): array;
+    public function attributesLabels(): array;
+    public function clearErrors(?string $attribute = null): void;
+    public function getAttributes(): array;
```

Comment:
What kind of differences of this method and `attributes`?

Reply:
attributes retrieve propertys with typehints.